Custom config not loaded properly

When linting with custom config file while `.swiftlint.yml` exists in current directory, `disabled_rules` are loaded from `.swiftlint.yml` instead of passed config file.

## How to reproduce (0.18.1):
In folder containing
```
# .swiftlint.yml
# empty
```
```
# .swiftlint-ci.yml
disabled_rules:
  - todo
```
```
// Foo.swift
// TODO
```
Run `swiftlint lint --config .swiftlint-ci.yml`:

>Loading configuration from '.swiftlint-ci.yml'
Linting Swift files in current working directory
Linting 'Foo.swift' (1/1)
Foo.swift:1:4: warning: Todo Violation: TODOs should be avoided. (todo)
Done linting! Found 1 violation, 0 serious in 1 file.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -64,6 +64,12 @@
   [#977](https://github.com/realm/SwiftLint/issues/977)
   [#1614](https://github.com/realm/SwiftLint/issues/1614)
 
+* Fix incorrectly using configuration files named `.swiftlint.yml` when they are
+  located in the same directory as a differently-named, user-provided custom
+  configuration file.  
+  [JP Simard](https://github.com/jpsim)
+  [#1531](https://github.com/realm/SwiftLint/issues/1531)
+
 ## 0.19.0: Coin-Operated Machine
 
 ##### Breaking

--- a/Source/SwiftLintFramework/Models/Configuration.swift
+++ b/Source/SwiftLintFramework/Models/Configuration.swift
@@ -313,6 +313,10 @@ private func warnAboutDeprecations(configurationDictionary dict: [String: Any],
 
 extension Configuration {
     fileprivate func configuration(forPath path: String) -> Configuration {
+        if path == rootPath {
+            return self
+        }
+
         let pathNSString = path.bridge()
         let configurationSearchPath = pathNSString.appendingPathComponent(Configuration.fileName)
 

--- a/Tests/SwiftLintFrameworkTests/ConfigurationTests.swift
+++ b/Tests/SwiftLintFrameworkTests/ConfigurationTests.swift
@@ -153,6 +153,12 @@ class ConfigurationTests: XCTestCase {
         return configuration
     }
 
+    private var projectMockConfig0CustomPath: Configuration {
+        var configuration = Configuration(path: projectMockYAML0CustomPath, optional: false, quiet: true)
+        configuration.rootPath = projectMockPathLevel0
+        return configuration
+    }
+
     fileprivate var projectMockConfig2: Configuration {
         return Configuration(path: projectMockYAML2, optional: false, quiet: true)
     }
@@ -189,6 +195,14 @@ class ConfigurationTests: XCTestCase {
     func testLevel3() {
         XCTAssertEqual(projectMockConfig0.configuration(for: File(path: projectMockSwift3)!),
                        projectMockConfig0.merge(with: projectMockConfig2))
+    }
+
+    // MARK: - Testing Custom Configuration File
+
+    func testCustomConfiguration() {
+        let file = File(path: projectMockSwift0)!
+        XCTAssertNotEqual(projectMockConfig0.configuration(for: file),
+                          projectMockConfig0CustomPath.configuration(for: file))
     }
 
     // MARK: - Testing Rules from config dictionary
@@ -282,6 +296,10 @@ fileprivate extension XCTestCase {
 
     var projectMockYAML0: String {
         return projectMockPathLevel0.stringByAppendingPathComponent(Configuration.fileName)
+    }
+
+    var projectMockYAML0CustomPath: String {
+        return projectMockPathLevel0.stringByAppendingPathComponent("custom.yml")
     }
 
     var projectMockYAML2: String {

--- a/Tests/SwiftLintFrameworkTests/Resources/ProjectMock/custom.yml
+++ b/Tests/SwiftLintFrameworkTests/Resources/ProjectMock/custom.yml
@@ -1,0 +1,8 @@
+disabled_rules:
+  - force_cast
+included:
+  - "only a few things"
+excluded:
+  - "the place where i committed many coding sins"
+line_length: 10000000000
+reporter: "json"

